src/sdkBase.ts

```diff
@@ -174,33 +188,109 @@ export abstract class SdkBase {
   }
 
   private initInternal(opts: SdkOptions | unknown) {
-    if (this._initComplete) {
-      throw new TypeError('Calling init() more than once is not allowed');
-    }
     if (!isSDKOptionsOrThrow(opts))
       throw new TypeError(`Options provided to ${this._product.name} init couldn't be validated.`);
 
-    storeConfig(opts, this._product);
+    if (this._hasAborted) {
+      throw new TypeError('Calling init() once aborted or disconnected is not allowed');
+    }
 
-    this._opts = opts;
-    this._storageManager = new StorageManager(
-      { ...opts },
-      this._product.cookieName,
-      this._product.localStorageKey
-    );
-    this._apiClient = new ApiClient(opts, this._product.defaultBaseUrl, this._product.name);
-    this._tokenPromiseHandler.registerApiClient(this._apiClient);
+    if (this.isInitialized()) {
+      this.setInitComplete(false);
```

Comment:
I don't think we need to set this to false - it's mainly used so we can throw an exception if one of our functions that requires config is called before init is called. This function isn't async, so there's not really a possibility of anything calling the SDK while we're updating the config.